lib/versioning/docker/index.js

```diff
@@ -1,8 +1,96 @@
-const semver = require('../semver');
+function parse(version) {
+  const [prefix, suffix] = version.split('-');
+  const release = prefix.split('.').map(Number);
+  if (release.some(Number.isNaN)) {
+    return null;
+  }
+  return { release, suffix: suffix || '' };
+}
 
-const isValid = input => semver.isVersion(input);
+function compare(version1, vervion2) {
+  const parsed1 = parse(version1);
+  const parsed2 = parse(vervion2);
+  const length = Math.max(parsed1.release.length, parsed2.release.length);
+  for (let i = 0; i < length; i += 1) {
+    const part1 = parsed1.release[i];
+    const part2 = parsed2.release[i];
+    // shorter is bigger 2.1 > 2.1.1
+    if (part1 === undefined) {
+      return 1;
+    }
+    if (part2 === undefined) {
+      return -1;
+    }
+    if (part1 !== part2) {
+      return part1 - part2;
+    }
+  }
+  // equals
+  return parsed2.suffix.localeCompare(parsed1.suffix);
```

Comment:
This is for cases where the suffix is like `alpha.1` and `beta.2`? not for `-alpine`?